### PR TITLE
Fix Bug 1410828 - Add a test to verify that the next top site in queue replaces a dismissed top site

### PR DESCRIPTION
**Pull request by icrisan**
Bug 1410828 - Add a test to verify that the next top site in queue replaces a dismissed top site

### CODE REVIEW

**Comment by sarracini** [comment]
@icrisan i had to revert your PR #4026, since it was failing tests when run in the suite. It seemed to not be clearing the pinned site after it pins it. I think a similar approach as what you're doing here would fix it, or perhaps even just manually unpinning it at the end of the test. Would you be able to open up a PR again for #4026 with a fix for that? Thank you!

File: system-addon/test/functional/mochitest/browser_topsites_contextMenu_options.js [hunk 1]
```diff
@@ -25,3 +22,34 @@ test_newtab({
...
+
+    // Need to wait for dismiss action.
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-sites-list li:first-child a").getAttribute("href") === secondTopSite,
+      "First topsite was dismissed");
+
+    defaultTopSitesNumber = content.document.querySelector(".top-sites-list").querySelectorAll("[class=\"top-site-outer\"]").length;
+    Assert.equal(defaultTopSitesNumber, 5, "5 top sites are displayed after one of them is dismissed");
+  },
+  async after() {
+    await SpecialPowers.pushPrefEnv({set: [["browser.newtabpage.blocked", ""]]});
```
Comment (Mardak) [inline system-addon/test/functional/mochitest/browser_topsites_contextMenu_options.js]: This doesn't seem right to call `SpecialPowers` here. There's already a `pushPrefs` globally, but the `before` should actually be using the scoped one instead. The scoped one cleans up after a test runs whereas the global one cleans up only when the test file is completed I believe.

File: system-addon/test/functional/mochitest/browser_topsites_contextMenu_options.js [hunk 2]
```diff
@@ -25,3 +22,34 @@ test_newtab({
...
+    await new Promise(resolve => NewTabUtils.undoAll(resolve));
+    await setDefaultTopSites();
```
Comment (Mardak) [inline system-addon/test/functional/mochitest/browser_topsites_contextMenu_options.js]: As noted in the previous comment, fixing up `setDefaultTopSites` to use the scoped one might avoid needing to do this here. If we do need it here for `--verify`, we should figure out why it's needed to figure out a better way to not need it.

File: system-addon/test/functional/mochitest/browser_topsites_contextMenu_options.js (hunk 1 above)
Comment (Mardak) [reply system-addon/test/functional/mochitest/browser_topsites_contextMenu_options.js]: Sure, but using the scoped one from `before` should clear that pref when the test finishes.

**Comment by piatra** [comment]
Updated.

**Comment by as-pine-proxy** [comment]
[Treeherder: 97e5ad49b32f5c5b09f5432aab367543af01eaac](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=97e5ad49b32f5c5b09f5432aab367543af01eaac)

**Pull request merged**